validator/specs/lib/validateCmptSpec.js

```diff
@@ -0,0 +1,191 @@
+    });
+
+    it('validates a cmpt tile with a valid b3dm inner tile', function() {
+        var innerB3dm = createB3dmTile();
+        var cmptTile = createCmptTile([innerB3dm]);
+
+        expect(validateCmpt(cmptTile).result).toBe(true);
+    });
+
+    it('validates a cmpt tile with a valid i3dm inner tile', function() {
+        var innerI3dm = createI3dmTile(0);
+        var cmptTile = createCmptTile([innerI3dm]);
+
+        expect(validateCmpt(cmptTile).result).toBe(true);
+    });
+
+    it('validates a cmpt tile with a valid pnts inner tile', function() {
+        var innerPnts = createPntsTile();
+        var cmptTile = createCmptTile([innerPnts]);
+
+        expect(validateCmpt(cmptTile).result).toBe(true);
+    });
+
+    it('validates a cmpt tile with a valid combinations of inner tiles', function() {
+        expect(validateCmpt(createCmptCombination()).result).toBe(true);
+    });
+
+    it('returns false if the cmpt has an invalid b3dm inner tile', function() {
+        var innerB3dm = createB3dmTile();
+        innerB3dm.writeUInt32LE(5, 4); // version
+        var cmptTile = createCmptTile([innerB3dm]);
+
+        expect(validateCmpt(cmptTile).result).toBe(false);
+    });
+
+    it('returns false if the cmpt has an invalid i3dm inner tile', function() {
+        var innerI3dm = createI3dmTile(15);
+        var cmptTile = createCmptTile([innerI3dm]);
+
+        expect(validateCmpt(cmptTile).result).toBe(false);
+    });
+
+    it('returns false if the cmpt has an invalid pnts inner tile', function() {
+        var innerPnts = createPntsTile();
+        innerPnts.writeUInt32LE(5, 4); // version
+        var cmptTile = createCmptTile([innerPnts]);
+
+        expect(validateCmpt(cmptTile).result).toBe(false);
+    });
+
+    it('returns false if the cmpt contains any invalid inner tiles', function() {
+        expect(validateCmpt(createCmptInvalidCombination()).result).toBe(false);
+    });
+});
+
+function createB3dmTile() {
+    var b3dmTile = new Buffer(b3dmHeaderSize);
+    b3dmTile.write('b3dm', 0); // magic
+    b3dmTile.writeUInt32LE(1, 4); // version
+    b3dmTile.writeUInt32LE(b3dmHeaderSize, 8); // byteLength
+
+    return b3dmTile;
+}
+
+function createI3dmTile(gltfFormat) {
+    var i3dmTile = new Buffer(i3dmHeaderSize);
+    i3dmTile.write('i3dm', 0); // magic
+    i3dmTile.writeUInt32LE(1, 4); // version
+    i3dmTile.writeUInt32LE(i3dmHeaderSize, 8); // byteLength
+    i3dmTile.writeUInt32LE(0, 12); // featureTableJSONByteLength
+    i3dmTile.writeUInt32LE(0, 16); // featureTableBinaryByteLength
+    i3dmTile.writeUInt32LE(0, 20); // batchTableJSONByteLength
+    i3dmTile.writeUInt32LE(0, 24); // batchTableBinaryByteLength
+    i3dmTile.writeUInt32LE(gltfFormat, 28); // gltfFormat: 0 - url
+
+    return i3dmTile;
+}
+
+function createCmptTile(tiles) {
+    var byteLength = cmptHeaderSize;
+    for(var i = 0; i < tiles.length; i++) {
+        byteLength += tiles[i].readUInt32LE(8);
+    }
```

Comment:
No need to find the byte length and send it to `Buffer.concat`. `Buffer.concat(tiles)` is fine.